src/test/java/org/opensearch/security/filter/RestPathMatchesTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.filter;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class RestPathMatchesTest {
+    Method restPathMatches;
+    SecurityRestFilter securityRestFilter;
+
+    @Before
+    public void setUp() throws NoSuchMethodException {
+        securityRestFilter = mock(SecurityRestFilter.class);
+        restPathMatches = SecurityRestFilter.class.getDeclaredMethod("restPathMatches", String.class, String.class);
+        restPathMatches.setAccessible(true);
+    }
+
+    @Test
+    public void testExactMatch() throws InvocationTargetException, IllegalAccessException {
+        String requestPath = "_plugins/security/api/x/y";
+        String handlerPath = "_plugins/security/api/x/y";
+        assertTrue((Boolean) restPathMatches.invoke(securityRestFilter, requestPath, handlerPath));
+    }
+
+    @Test
+    public void testPartialMatch() throws InvocationTargetException, IllegalAccessException {
+        String requestPath = "_plugins/security/api/x/y";
+        String handlerPath = "_plugins/security/api/x/z";
+        assertFalse((Boolean) restPathMatches.invoke(securityRestFilter, requestPath, handlerPath));
+    }
+
+    @Test
+    public void testNamedParamsMatch() throws InvocationTargetException, IllegalAccessException {
```

Comment:
I was a bit confused by this test case that where to get the actual value of `{id}`. Or is the method `restPathMatches` seems to understand these placeholders, and considers "123" in the requestPath to match "{id}" in the handlerPath?

Reply:
yes. Parameter {id} will resolve to 123 for this request path which is what we want. {id} acts as a placeholder in the handler for any actual parameters passed in the request.